internal/cache/persistent_lru_cache.go

```diff
@@ -225,27 +200,14 @@ func (c *PersistentCache) Put(ctx context.Context, key string, data gather.Bytes
 	c.listCacheMutex.Lock()
 	defer c.listCacheMutex.Unlock()
 
-	// opportunistically cleanup cache before the PUT if we can
-	if c.isCacheFullLocked() {
-		c.listCacheCleanupLocked(ctx)
-		// Do not add more things to cache if it remains full after cleanup. We
-		// MUST NOT go over the specified limit for the cache space to avoid
-		// snapshots/restores from getting affected by the cache's storage use.
-		if c.isCacheFullLocked() {
-			// Limit warnings to one per minute max.
-			if clock.Now().Sub(c.lastCacheWarning) > 10*time.Minute {
-				c.lastCacheWarning = clock.Now()
-
-				log(ctx).Warnf("Cache is full, unable to add item into '%s' cache.", c.description)
-			}
-
-			return
-		}
-	}
+	c.storageProtection.Protect(key, data, &protected)
+
+	// make sure the cache has enough room for the new item including any protection overhead.
+	c.pendingWriteBytes += int64(protected.Length())
+	c.listCacheCleanupLocked(ctx)
```

Comment:
What if the cache remains full even after running `listCacheCleanupLocked()` due to `MinSweepAge` ?